src/components/ConfigProvider/Theming/Theming.types.ts

```diff
@@ -15,23 +15,198 @@ export type OcThemeNames =
 export type ThemeName = OcThemeNames | OcCustomThemeName;
 
 export interface OcBaseTheme {
-    primaryColor?: Color;
-    disruptiveColor?: Color;
-    textColor?: Color;
-    textColorSecondary?: Color;
-    textColorInverse?: Color;
-    backgroundColor?: Color;
-    successColor?: Color;
-    warningColor?: Color;
-    infoColor?: Color;
-    errorColor?: Color;
+    primaryColor?: Value;
+    disruptiveColor?: Value;
+    textColor?: Value;
+    textColorSecondary?: Value;
+    textColorInverse?: Value;
+    backgroundColor?: Value;
+    successColor?: Value;
+    warningColor?: Value;
+    infoColor?: Value;
+    errorColor?: Value;
+    styles?: OcBaseStyles;
+}
+
+export interface OcGlobalStyles {
+    /**
+     * Component border-radius
+     */
+    borderRadius?: Value;
+    /**
+     * Component border-width
+     */
+    borderThickness?: Value;
+    /**
+     * Component font-size
+     */
+    fontSize?: Value;
+    /**
+     * Component margin
+     */
+    margin?: Value;
+    /**
+     * Component padding
+     */
+    padding?: Value;
+}
+
+export interface OcButtonStyles extends OcGlobalStyles {
+    /**
+     * Component background color
+     */
+    backgroundColor?: Value;
+    /**
+     * Component background active color
+     */
+    backgroundActiveColor?: Value;
+    /**
+     * Component background hover color
+     */
+    backgroundHoverColor?: Value;
+    /**
+     * Component background focus color
+     */
+    backgroundFocusColor?: Value;
+    /**
+     * Component background visited color
+     */
+    backgroundVisitedColor?: Value;
+    /**
+     * Component text color
+     */
+    textColor?: Value;
+    /**
+     * Component text active color
+     */
+    textActiveColor?: Value;
+    /**
+     * Component text hover color
+     */
+    textHoverColor?: Value;
+    /**
+     * Component text focus color
+     */
+    textFocusColor?: Value;
+    /**
+     * Component text visited color
+     */
+    textVisitedColor?: Value;
+}
+
+export interface OcDialogStyles extends OcButtonStyles {}
+
+export interface OcDropdownStyles extends OcButtonStyles {}
+
+export interface OcIconStyles extends OcButtonStyles {}
+
+export interface OcInfoBarStyles extends OcButtonStyles {}
+
+export interface OcInputStyles extends OcButtonStyles {}
+
+export interface OcListStyles extends OcButtonStyles {}
+
+export interface OcMenuStyles extends OcButtonStyles {}
+
+export interface OcModalStyles extends OcDialogStyles {}
```

Comment:
we can refine where components should derive styles, for example Modal should extend Dialog, et cetera.